activity_browser/ui/tables/models/base.py

```diff
@@ -420,18 +420,24 @@ class ABSortProxyModel(QSortFilterProxyModel):
 
     See this for context: https://github.com/LCA-ActivityBrowser/activity-browser/pull/1151
     """
-    def lessThan(self, left, right) -> bool:
+    def lessThan(self, left: QModelIndex, right: QModelIndex) -> bool:
         """Override to sort actual data, expects `left` and `right` are comparable.
 
         If `left` and `right` are not the same type, we check if numerical and empty string are compared, if that is the
         case, we assume empty string == 0.
         Added this case for: https://github.com/LCA-ActivityBrowser/activity-browser/issues/1215"""
-        left_data = self.sourceModel().data(left, 'sorting')
-        right_data = self.sourceModel().data(right, 'sorting')
```

Comment:
If you don't keep the `sorting` argument, the sort is done based on what is in the view, not the actual data (which is the reason `QSortFilterProxyModel` was subclassed in the first place). 

e.g. timestamps in a human readable format are not properly sorted if this argument is removed (#1151)


please restore.

Reply:
> It's not normally good practice to be passing strings like this

Ok, what _is_ good practice in this case? Happy to hear suggestions.

Reply:
The `'sorting'` string is passed as `role` argument, Qt just doesn't provide a role for what we are doing, so we created a custom one like this. I could have written a custom role, but just passing a string is less of a mess IMO